src/utils/utils_load_library.h

```diff
@@ -39,22 +39,34 @@ static inline void *util_open_library(const char *filename) {
 static inline int util_close_library(void *handle) {
     // If the FreeLibrary function succeeds, the return value is nonzero.
     // If the FreeLibrary function fails, the return value is zero.
-    return (FreeLibrary(handle) == 0);
+    return (FreeLibrary((HMODULE)handle) == 0);
 }
 
-static inline void *util_get_symbol_addr(void *handle, const char *symbol) {
-    return GetProcAddress(handle, symbol);
+static inline void *util_get_symbol_addr(void *handle, const char *symbol,
+                                         const char *libname) {
+    if (handle == NULL) {
+        if (libname == NULL) {
+            return NULL;
+        }
+        handle = GetModuleHandle(libname);
+    }
+    return GetProcAddress((HMODULE)handle, symbol);
 }
 
 #else /* Linux */
 
 static inline void *util_open_library(const char *filename) {
-    return dlopen(filename, RTLD_LAZY);
+    // NOTE that we use RTLD_GLOBAL which add all loaded symbols to the
+    // global symbol table. These symbols would be used by the Level Zero
+    // provider later
+    return dlopen(filename, RTLD_GLOBAL | RTLD_LAZY);
```

Comment:
I'm not sure if we need GLOBAL. Would CI work without this change? Also, please remove 2nd sentence about Level Zero from this comment as this function is used for any library.

Reply:
OK I think the best wat would be to add additional "flags" argument to this function that should be ORed with RTLD_LAZY. Then we could set GLOBAL only in calls from provider_level_zero.cpp

Reply:
I propose something like this